torch/_functorch/_aot_autograd/graph_capture.py

```diff
@@ -74,12 +98,13 @@ def _detach_and_copy_item_memo(t):
 
 
 def aot_dispatch_base_graph(
-    flat_fn,
-    flat_args: list[Tensor],
+    flat_fn: TraceFn,
+    flat_args: list[FxValue],
+    flat_args_descs: list[AOTInput],
     aot_config: AOTConfig,
     *,
     fw_metadata: ViewAndMutationMeta,
-) -> tuple[torch.fx.GraphModule, list[Any], Optional[SubclassMeta]]:
+) -> tuple[torch.fx.GraphModule, list[FxValue], list[AOTInput], Optional[SubclassMeta]]:
```

Comment:
I feel like, in a perfect world, you'd be able to have different types for the input descriptors and output descriptors here, since theroteically you've elaborated the AOTInputs in some way

Reply:
Presumably this function changes the AOTInput passed into some way, my question was should it change the type to a more specific type of AOTInput?

But I think the answer is the input AOTInput can be some set of AOTInput subtypes, and the output AOTInput can be a different set of AOTInput subtypes, and there's no reason to be that specific about the type I guess.

Reply:
So, in fact, I discovered (while working on another CR comment from you) that it might be possible to have some sort of staging from DifferentiableAOTInput -> AOTInput, reflecting the fact that all inputs are differentiable before we do the autograd transforms, and then non-differentiable afterwards. I got three files in before I gave up on the refactor here, so I leave it to future work.